src/main/java/software/amazon/qldb/QldbDriverImpl.java

```diff
@@ -94,6 +95,12 @@ protected QldbDriverImpl(String ledgerName,
         this.isClosed = new AtomicBoolean(false);
         this.readAhead = readAhead;
         this.executorService = executorService;
+        int iseRetryLimit = maxConcurrentTransactions + 3;
+        if (maxConcurrentTransactions > Integer.MAX_VALUE - 3) {
```

Comment:
Wouldn't it be better to do the check before assigning maxConccurentTransactions +3? 

If going over the max value in l98 isn't problematic then it's fine, but just to be safe.

Reply:
Just a style preference - keeping as is.